### ISSUE TITLE
Is fix_imbalance = True working?

### ISSUE THREAD

**Issue by Turanga1** [pycaret/pycaret#1650]
Setting fix_imbalance=True doesn't appear to change anything. The training dataset after preprocessing is still imbalanced and classification metrics show very high accuracy results associated with guessing the majority class. Here's a simple example:

```from sklearn.datasets import make_classification
from pycaret.classification import *
import pandas as pd

# random dataset
X, y = make_classification(n_samples=10000, n_features=2, n_redundant=0,
        n_clusters_per_class=1, weights=[0.99], flip_y=0, random_state=1)
data = pd.DataFrame({'A':X[:,0], 'B':X[:,1], 'y':y})

# distribution in original dataset
counter = Counter(data['y'])
print(counter)

# PyCaret with fix_imbalance=True
clf1 = setup(data, target = 'y', session_id=123, log_experiment=True, fix_imbalance=True, experiment_name='app1')

# data after (?) fixing imbalance
print('SMOTE settings:', clf1[8])
print(clf1[33][2][0])
counter = Counter(clf1[33][2][1])
print(counter)
```

> SMOTE settings: 
> Pipeline(memory=None,
>          steps=[('fix_imbalance',
>                  SMOTE(k_neighbors=5, n_jobs=None, random_state=123,
>                        sampling_strategy='auto'))],
>          verbose=False)

> y_training Set:
> Counter({0: 6935, 1: 64})

**Comment by kilickursat** [comment]
Can you please elaborate resampling during per fold? Especially for under sampling with per fold. Thank you.

**Comment by Yard1** [comment]
The resampling is applied during model fitting for every fold (so that the changes are reflected in cross validation results) and during fitting of the model that is returned. Therefore, the changes are not visible in the train and test datasets you can obtain with `get_config`.

**Comment by kilickursat** [comment]
Thanks for your reply. How can I cite this information? Could you suggest to me a paper about it? Because I want to use this information in my paper.

**Comment by kilickursat** [comment]
In addition, why pycaret use train and test split with kfold together?
